[Isaac] iP

Isaaclhy00 opened the pull request:
### **Duke**

“Productivity is never an accident. It is always the result of a commitment to excellence, intelligent planning, and focused effort.” – Paul J. Meyer ([source](https://dansilvestre.com/productivity-quotes))

Duke tracks your tasks for you so your mind is free to focus on the tasks at hand. It's,

* text-based
* intuitive
* ~~quick~~  LIGHTNING FAST in response time
* All you need to do is,

1. Download it from [here](https://github.com/Isaaclhy00/ip.gitl).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉
# And it is FREE!

Features:

- [x] Managing tasks
- [x] Managing deadlines 
- [x] Organising tasks
- [  ]Reminders (coming soon)
If you Java programmer, you can use it to practice Java too. Here's the main method:

`public class Main {
    public static void main(String[] args) {
        new Duke(data/tasks.txt).run();
    }
}`

Commit:
Add Gradle support

Isaaclhy00 committed:
level 1-3 I’mplmented

Isaaclhy00 committed:
level 1-4 Implemented

Isaaclhy00 committed:
Level 1-6 implmented

Isaaclhy00 committed:
Implemented level 7

Isaaclhy00 committed:
Implemented Level 8

Isaaclhy00 committed:
Merge branch 'add-gradle-support'

Isaaclhy00 committed:
Implemented More OOP

Isaaclhy00 committed:
Implemented JUnit

vvidday commented on src/main/java/Duke/commands/Command.java:
```diff
@@ -0,0 +1,26 @@
+
+import java.time.format.DateTimeFormatter;
+
+public abstract class Command {
```
I like the use of an abstract class here, as well as your great organization in the `commands` package!

vvidday commented on src/main/java/Duke/storage/Storage.java:
```diff
@@ -0,0 +1,66 @@
+        }
+    }
+
+    public void store(TaskList toStore) {
```
Maybe you could consider writing javadocs for these more complicated public facing methods, to give others a clearer picture of what they do?

vvidday commented on src/main/java/Duke/commands/AddTodoCommand.java:
```diff
@@ -0,0 +1,32 @@
+
+    @Override
+    public void execute(TaskList taskList, UI ui, Storage storage) {
+        String taskDesc = input.substring(5);
```
Perhaps a named constant instead of a magic number would make the code here (and other parts) clearer?

vvidday commented on src/main/java/Duke/commands/DeleteTaskCommand.java:
```diff
@@ -0,0 +1,28 @@
+    private final int taskID;
+
+    public DeleteTaskCommand(String input) throws DukeException {
+        if (!checkValid(input)) throw new DukeException(" ☹ OOPS!!! Please enter task to delete.");
```
Should the conditional here be wrapped in curly brackets?

vvidday commented on src/main/java/Duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,66 @@
+        return taskList.size() == 0;
+    }
+
+    public int size() {
```
Should this method name be a verb? e.g. `getSize()`

vvidday commented on src/main/java/Duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,66 @@
+
+    public void printList() {
+        try {
+            if (isEmpty()) {
+                throw new DukeException(" ☹ OOPS!!! Seems like your list is empty.");
```
I like that you thought of & handled this edge case!

vvidday reviewed (commented):
Overall, your code looks good and I really like how you organized it nicely into different packages, especially `commands`! There's a couple of minor code quality / coding standard issues below, but other than that, LGTM!

marcusczh commented on src/main/java/Duke/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,44 @@
+package Duke.commands;
```
You may want to consider changing names representing packages to all lowercase, to conform to the coding standard.

marcusczh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+        }
+    }
+
+    public void run() {
```
I really like how all the detailed functionalities of Duke.run() are abstracted further into classes.

marcusczh commented on src/main/java/Duke/storage/Storage.java:
```diff
@@ -0,0 +1,66 @@
+package Duke.storage;
+
+import Duke.tasks.*;
```
You may want to consider avoiding wildcard imports

marcusczh commented on src/main/java/Duke/parser/Parser.java:
```diff
@@ -0,0 +1,40 @@
+import Duke.commands.*;
+import Duke.exceptions.DukeException;
+
+public class Parser {
```
I like how you abstracted the functionalities here into separate Command classes

marcusczh commented on src/main/java/Duke/commands/MarkTaskCommand.java:
```diff
@@ -0,0 +1,28 @@
+
+public class MarkTaskCommand extends Command {
+
+    private final int taskID;
```
I think taskID can be written as taskId instead to conform to the camelCase standard

marcusczh reviewed (commented):
Other than some minor code standard issues, the overall code looks good and is ready to merge! I really like the use of OOP to abstract out the common functionalities as well!

Isaaclhy00 committed:
Added JavaDocs

Isaaclhy00 committed:
Implemented Find

Isaaclhy00 committed:
Updated Duke.java

Isaaclhy00 committed:
Implemented Checkstyle

Isaaclhy00 committed:
Implement JavaFX for GUI

Duke uses terminal to receive inputs and display outputs. This interface is ugly and not user friendly.

New GUI was implemented to make Duke appear in chat form, much more friendly.

Isaaclhy00 committed:
Implemented Undo

Isaaclhy00 committed:
Merge pull request #1 from Isaaclhy00/master

CodeQuality Branch

Isaaclhy00 committed:
Implemented C-Undo

Undo has some bugs

It does not work as intended after some testing

The implementation was changed to fix the bugs

Isaaclhy00 committed:
Merge branch 'branch-A-CodeQuality'

Isaaclhy00 committed:
Improved GUI

GUi is lacktuster

Made several improvements (added profile picture, etc...)

Fixed several bugs from previous code

Isaaclhy00 committed:
Update README.md

Isaaclhy00 committed:
Fix Jar file

Isaaclhy00 committed:
Update build.gradle

Isaaclhy00 committed:
Revert "Update build.gradle"

This reverts commit 5f7ef7b0690c9c3a73b8449cb66b7d43220fc47d.

Isaaclhy00 committed:
Update build.gradle

Isaaclhy00 committed:
Update build.gradle

Isaaclhy00 committed:
Updated duke.jar

Isaaclhy00 committed:
Update build.gradle

Isaaclhy00 committed:
Update build.gradle

Isaaclhy00 committed:
Update build.gradle

Isaaclhy00 committed:
Final Version (Fixed jar)
